src/coreclr/jit/gentree.cpp

```diff
@@ -21308,6 +21308,158 @@ GenTree* Compiler::gtNewSimdCeilNode(var_types type, GenTree* op1, CorInfoType s
+                case TYP_ULONG:
+                    for (int i = 0; i < 16; i++)
+                    {
+                        tbl->gtSimdVal.i32[i] = 0x08080088;
+                    }
+                    break;
+
+                case TYP_INT:
+                case TYP_LONG:
+                    for (int i = 0; i < 16; i++)
+                    {
+                        tbl->gtSimdVal.i32[i] = 0x00000088;
+                    }
+                    break;
+
+                default:
+                    unreached();
+            }
+            break;
+
+        default:
+            unreached();
+    }
+    return tbl;
+}
+
+GenTree* Compiler::gtNewSimdCvtNode(var_types      type,
+                                    GenTree*       op1,
+                                    NamedIntrinsic hwIntrinsicID,
+                                    CorInfoType    simdTargetBaseJitType,
+                                    CorInfoType    simdSourceBaseJitType,
+                                    unsigned       simdSize)
+{
+    assert(varTypeIsSIMD(type));
+    assert(getSIMDTypeForSize(simdSize) == type);
+    assert(op1 != nullptr);
+    assert(op1->TypeIs(type));
+
+    var_types simdSourceBaseType = JitType2PreciseVarType(simdSourceBaseJitType);
+    var_types simdTargetBaseType = JitType2PreciseVarType(simdTargetBaseJitType);
+    assert(varTypeIsArithmetic(simdSourceBaseType));
+    assert(varTypeIsArithmetic(simdTargetBaseType));
+    assert(varTypeIsFloating(simdSourceBaseType));
+    assert(varTypeIsIntegral(simdTargetBaseType));
+
+    assert(IsBaselineSimdIsaSupportedDebugOnly());
+    assert(IsBaselineVector512IsaSupportedDebugOnly() ||
+           ((simdTargetBaseType == TYP_INT) && ((simdSize == 16 && compIsaSupportedDebugOnly(InstructionSet_SSE41)) ||
+                                                (simdSize == 32 && compIsaSupportedDebugOnly(InstructionSet_AVX)))));
+
+    GenTree* fixupVal;
+
+    if (IsBaselineVector512IsaSupportedOpportunistically())
+    {
+        GenTreeVecCon* tbl = gtCvtCtrlTbl(type, simdSourceBaseType, simdTargetBaseType, simdSize);
+
+        GenTree* op1Clone = fgMakeMultiUse(&op1);
+
+        // run vfixupimmsd base on table and no flags reporting
+        fixupVal = gtNewSimdHWIntrinsicNode(type, op1, op1Clone, tbl, gtNewIconNode(0), NI_AVX512F_Fixup,
+                                            simdSourceBaseJitType, simdSize);
+    }
+    else
+    {
+        GenTree* op1Clone1 = fgMakeMultiUse(&op1);
+        GenTree* op1Clone2 = fgMakeMultiUse(&op1);
+        GenTree* mask1     = gtNewSimdCmpOpNode(GT_EQ, type, op1, op1Clone1, simdSourceBaseJitType, simdSize);
+        fixupVal           = gtNewSimdBinOpNode(GT_AND, type, op1Clone2, mask1, simdSourceBaseJitType, simdSize);
+    }
+
+    if (varTypeIsSigned(simdTargetBaseType))
+    {
+        GenTree* maxVal;
+        GenTree* maxValDup;
+        if (varTypeIsLong(simdTargetBaseType))
+        {
+            long long actualMaxVal = INT64_MAX;
```

Comment:
nit, it helps to be more explicit about the type to avoid differences across compilers:
```suggestion
            int64_t actualMaxVal = INT64_MAX;
```

Reply:
Made this change.